server/app/controllers/fusor/api/v2/deployments_controller.rb

```diff
@@ -75,8 +75,15 @@ def deploy
         task = async_task(::Actions::Fusor::Deploy,
                           @deployment,
                           params[:skip_content])
+        @deployment.foreman_task_uuid = task.id
+        @deployment.save
       end
-      respond_for_async :resource => task
+
+      if !task
```

Comment:
Ruby loves it's one liners...

```ruby
raise "ManageManifest task failed" unless task
```

Also, always prefer &&, II, ! over and, or, and not. The precedence is very low and results in often unexpected results.